SU2_CFD/src/variables/CNEMOEulerVariable.cpp

```diff
@@ -295,23 +300,23 @@ bool CNEMOEulerVariable::Cons2PrimVar(su2double *U, su2double *V,
     } else {
       V[TVE_INDEX]   = T[1];
     }
-  }
-  else {
+  } else {
+    //TODO: can e-modes/vibe modes be active?

```

Comment:
no, they can't. but I don't think there's a TO-DO here.

what we did here was: for monoatomic species the concept of Tve does not exist, and if we try to compute it as usual, it gives trouble... so what we did was simply always set it to the freeestream value, as a dummy value, since it needs to be populated with something, but a 'something' that doesn't disturb the rest, so we thought freestream would do the job (even if Tve is not set in the config, which theoretically shouldn't be for monoatomic species, we have the nice 288.15 value as default)

Reply:
I gotcha.  I wasn't sure if a gas could have its electron state excited (without ejecting electrons) in a manner that impacted the non-equilibrium models.   But I can remove.

Reply:
I see you thinking, but I think - might be wrong here - that electron do not have Tve. Tve is electronic-vibrational, referring to electronic states of atoms and molecules, and vibrational states of molecules. electronic state vs. electron. I think electron either have it's own temperature, or go with translational temperature - but I am not sure, maybe @jtneedels can help here

Reply:
So my understanding is that the a monatomic species can have Tve. Since monatomic species don't have vibrational energy, that contribution to the Eve (and therefore Tve) is zero. However, variation in the configuration of electrons in the electronic energy levels does contribute to change in Tve. Equation 2.33 in Sean Copeland's thesis gives the definition of vibrational-electronic energy for general atomic/molecular species (including monatomic) as well as electrons. Using this framework, electrons are defined as only being able to carry Tve. So even if electrons are not being ejected, we still need to account for the excitation of electronic energy levels in the monatomic species.

Reply:
thanks. I am following, but 2 things:
1- I understand that electronic levels will be there for monoatomic species. but if, in the 2t model, vibrational and electronic go together, it means that if the species is monoatomic, e_ve=0 => Tve = 0 despite. so in that sense, the code should remain as it is. correct me if I'm wrong pls
2- if you agree with 1, what we could do instead is set Tve = 0 , instead of freestream, and see if it doesn't cause any trouble

Reply:
I am not sure that we can claim (1) directly from the assumptions of the 2T model. Mode separability implies that rho * e_ve = \sum{rho_s * (e_vib,s + e_el,s}). So even if the contribution of e_vib,s is zero, the net contribution to electronic energy from a monatomic species can be non-zero if it has a non-zero electronic energy contribution. My understanding is that the assumption of equilibrium between the vibrational and electronic energy modes does not imply that if a singles species has no vibrational modes its e_ve contribution is automatically zero.

Reply:
you're right, I know that ahah. my brain stopped working there for a moment. Then this means we need to review this. I'll add it as a TO-DO for now, and maybe we can discuss with tomorrow with everyone, especially @fmpmorgado who is the one that introduced the monoatomic mixtures

Reply:
Yeah, that sounds great!